### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by chandrashekar-cohesity**
Addded documentation for new modules. Please take a look and confirm

### CODE REVIEW

File: docs/modules/cohesity_clone_vm.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+- [Examples](#examples)
+  - [Create a Test/Dev VM clone](#Create-a-Test/Dev-VM-clone)
+  - [Destroy a Test/Dev VM clone](#Destroy-a-Test/Dev-VM-clone)
+- [Parameters](#parameters)
+- [Outputs](#outputs)
+
+## Synopsis
+[top](#cohesity-clone-management---test&dev)
+
+The Ansible Module creates, modifies or deletes a Protection Policy for the Cohesity Cluster
```
Comment (anvesh-cohesity) [inline docs/modules/cohesity_clone_vm.md]: This module is for clone vm right, this one talks on protection policy. Please change this

File: docs/modules/cohesity_clone_vm.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+### Requirements
+[top](#cohesity-clone-management---test&dev)
+
+* Cohesity DataPlatform running version 6.0 or higher
```
Comment (anvesh-cohesity) [inline docs/modules/cohesity_clone_vm.md]: qualified for 6.3 or higher. Can you change it to 6.3 or higher

File: docs/modules/cohesity_clone_vm.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+  - Currently, the Ansible Module requires Full Cluster Administrator access.
+  - When using the default download location, the Cohesity agent installer is placed in `/tmp/<temp-dir`.  If your environment prevents the use of `/tmp` with a `noexec` option, then you must set an alternate location.
+
+## Syntax
+[top](#cohesity-clone-management---test&dev)
+
+```yaml
+- cohesity_clone_vm:
+    name: <Name of the Clone Task>
+    state: <Determines if the Clone Task should be present or absent from the host>
```
Comment (anvesh-cohesity) [inline docs/modules/cohesity_clone_vm.md]: what does host refer to here, is it Cohesity cluster?

File: docs/modules/cohesity_clone_vm.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    job_name: <Protection Group/Job name from where VM will be cloned>
+    view_name: <View name which will be cloned along with the VM>
```
Comment (anvesh-cohesity) [inline docs/modules/cohesity_clone_vm.md]: the view is not cloned here, please check the API doc. here is what this view is for "Specifies the name of the View where the cloned VMs are stored.". Pls change this

File: docs/modules/cohesity_clone_vm.md [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    backup_timestamp: <Specify point in time snapshot using this option>
```
Comment (anvesh-cohesity) [inline docs/modules/cohesity_clone_vm.md]: be more clear here "Cohesity Backup Run time. The formart should be YYYY-MM-DD:hh:mm. If not selected, the most recent backup is used"

File: docs/modules/cohesity_clone_vm.md [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+    environment: <Specify point in time snapshot using this option>
```
Comment (anvesh-cohesity) [inline docs/modules/cohesity_clone_vm.md]: environment doesn't talk of time, please change this

File: docs/modules/cohesity_clone_vm.md [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+    vm_name: <Name of the VMs that will be cloned>
```
Comment (anvesh-cohesity) [inline docs/modules/cohesity_clone_vm.md]: list of VM's names that will be cloned. The parameter is vm_names not vm_name

File: docs/modules/cohesity_clone_vm.md [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+    wait_for_job: <ASK_TEAM>
```
Comment (anvesh-cohesity) [inline docs/modules/cohesity_clone_vm.md]: wait for clone task, waits for wait minutes passed to module, default wait minutes is 30 mins

File: docs/modules/cohesity_clone_vm.md [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+    prefix: <Add prefix to cloned VM name >
+    suffix: <Add suffix to cloned VM name >
+    power_on: <Specify if you want cloned VM powered on or off>
+    network_conntected: <Specify if you want cloned VM connected to network or a detached network>
```
Comment (anvesh-cohesity) [inline docs/modules/cohesity_clone_vm.md]: Specifies whether the network should be left in disabled state. Attached network is enabled by default. Set this flag to true to disable it.

File: docs/modules/cohesity_clone_vm.md [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+    wait_minutes: <ASK_TEAM>
```
Comment (anvesh-cohesity) [inline docs/modules/cohesity_clone_vm.md]: wait time in minutes, time to wait for clone task

File: docs/modules/cohesity_clone_vm.md [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+    resource_pool: <ASK_TEAM>
```
Comment (anvesh-cohesity) [inline docs/modules/cohesity_clone_vm.md]: Specifies the resource pool where the cloned VMs are
attached.

File: docs/modules/cohesity_clone_vm.md [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+    password: password
+    name: CLONE_TASK_NAME
+    view_name: VIEW_NAME
+    environment: VMware
+    vm_names:
+        - VM_Name1
+        - VM_Name2
+    resource_pool: RESOURCE_POOL_NAME
+    state: present
+```
```
Comment (anvesh-cohesity) [inline docs/modules/cohesity_clone_vm.md:71]: Have you verified if this example works

File: docs/modules/cohesity_clone_vm.md [hunk 13]
```diff
@@ -0,0 +1,114 @@
...
+    password: password
+    name: CLONE_TASK_NAME
+    view_name: VIEW_NAME
+    environment: VMware
+    vm_names:
+        - VM_Name1
+        - VM_Name2
+    resource_pool: RESOURCE_POOL_NAME
+    state: absent
+```
```
Comment (anvesh-cohesity) [inline docs/modules/cohesity_clone_vm.md:90]: Have you verified if this works

File: docs/modules/cohesity_clone_vm.md [hunk 14]
```diff
@@ -0,0 +1,114 @@
...
+
+## Parameters
+[top](#cohesity-clone-management---test&dev)
+
+| Required | Parameters | Type | Choices/Defaults | Comments |
+| --- | --- | --- | --- | --- |
+| X | **name** | String | | Name of the Cloned VM. |
+|  | state | Choice | -**present**<br>-absent | Determines if the clone should be present or absent from the host.|
+| X | **job_name** | String | | Protection Group/Job name from where VM will be cloned. |
+| X | **view_name** | String | | View name which will be cloned along with the VM. |
```
Comment (anvesh-cohesity) [inline docs/modules/cohesity_clone_vm.md]: change the view name description to the one pointed above

File: docs/modules/cohesity_clone_vm.md [hunk 15]
```diff
@@ -0,0 +1,114 @@
...
+| | backup_timestamp | String | | Specify point in time snapshot using this option. |
```
Comment (anvesh-cohesity) [inline docs/modules/cohesity_clone_vm.md]: be more specific and the format if should be in

File: docs/modules/cohesity_clone_vm.md [hunk 16]
```diff
@@ -0,0 +1,114 @@
...
+| | environment | Choice | VMware | Select the source environment for cloning. |
+|   | vm_names | List | | Name of the VMs that will be cloned. |
+|   | wait_for_job | Boolean | True | ASK_TEAM|
```
Comment (anvesh-cohesity) [inline docs/modules/cohesity_clone_vm.md]: mentioned in previous comment

File: docs/modules/cohesity_clone_vm.md [hunk 17]
```diff
@@ -0,0 +1,114 @@
...
+|  | prefix | String | | Add prefix to cloned VM name. |
+|  | suffix |String | | Add suffix to cloned VM name. |
+|   | power_on | Boolean | True | Specify if you want cloned VM powered on or off.|
+|   | network_connected | Boolean | True | Specify if you want cloned VM powered on or off.|
+|   | wait_minutes | Integer | 30 | ASK_TEAM|
```
Comment (anvesh-cohesity) [inline docs/modules/cohesity_clone_vm.md]: mentioned in previous comment

File: docs/modules/cohesity_clone_vm.md [hunk 18]
```diff
@@ -0,0 +1,114 @@
...
+| X | **resource_pool** | String | | ASK_TEAM |
```
Comment (anvesh-cohesity) [inline docs/modules/cohesity_clone_vm.md]: mentioned in previous comment

File: docs/modules/cohesity_policy.md [hunk 19]
```diff
@@ -0,0 +1,121 @@
+# Cohesity Policy Management - Linux
+
+## Table of Contents
+- [Synopsis](#synopsis)
+- [Requirements](#requirements)
+- [Syntax](#syntax)
+- [Examples](#examples)
+  - Policy Management
+    - [Create a Protection Policy](#Create-a-Protection-Policy)
+    - [Modify a Protection Policy](#Modify-a-Protection-Policy)
```
Comment (anvesh-cohesity) [inline docs/modules/cohesity_policy.md]: modifying a policy is not supported

File: docs/modules/cohesity_policy.md [hunk 20]
```diff
@@ -0,0 +1,121 @@
...
+    - [Delete a Protection Policy](#Delete-a-Protection-Policy)
+- [Parameters](#parameters)
+- [Outputs](#outputs)
+
+## Synopsis
```
Comment (anvesh-cohesity) [inline docs/modules/cohesity_policy.md:15]: please update this doc based on the parameter descriptions I sent you

File: docs/tasks/clone_vm.md [hunk 21]
```diff
@@ -0,0 +1,181 @@
+# Task: Cohesity Clone VM 
+
+## Table of Contents
+- [Synopsis](#synopsis)
+- [Requirements](#requirements)
+- [Ansible Variables](#Ansible-Variables)
+- [Customize Your Playbooks](#Customize-your-playbooks)
+  - [Install the Cohesity Agent on Linux hosts](#Install-the-Cohesity-Agent-on-Linux-hosts)
```
Comment (anvesh-cohesity) [inline docs/tasks/clone_vm.md]: This talks on Agent installation, please change this and the ones below

File: docs/tasks/clone_vm.md [hunk 22]
```diff
@@ -0,0 +1,181 @@
...
+  - [Install the Cohesity Agent on Linux hosts using Root](#Install-the-Cohesity-Agent-on-Linux-hosts-using-Root)
+  - [Install the Cohesity Agent on Linux hosts using a custom download path](#Install-the-Cohesity-Agent-on-Linux-hosts-using-a-custom-download-path)
+- [How the Task Works](#How-the-Task-works)
+
+## Synopsis
+[top](#Cohesity-clone-vm)
+
+Use this task to clone a VM using Test/Dev workflow
```
Comment (anvesh-cohesity) [inline docs/tasks/clone_vm.md]: to create and destroy the clone, please add more details here

File: docs/tasks/clone_vm.md [hunk 23]
```diff
@@ -0,0 +1,181 @@
...
+
+#### How It Works
+- The tasks collects all the information (required and optional) and starts a clone task (if *state=present*).
+- The task can also destory a clone (*state=absent*) that was cloned using Test/Dev workflow.
+
+
+### Requirements
+[top](#Cohesity-clone-vm)
+
+* Cohesity DataPlatform running version 6.0 or higher
```
Comment (anvesh-cohesity) [inline docs/tasks/clone_vm.md]: qualified for 6.3 or higher

File: docs/tasks/clone_vm.md [hunk 24]
```diff
@@ -0,0 +1,181 @@
...
+                name: CLONE_TASK_NAME
+                view_name: VIEW_NAME
+                environment: VMware
+                vm_names:
+                    - VM_Name1
+                    - VM_Name2
+                resource_pool: RESOURCE_POOL_NAME
+                state: present
+```
+
```
Comment (anvesh-cohesity) [inline docs/tasks/clone_vm.md:114]: verified if this playbook works?

File: docs/tasks/clone_vm.md [hunk 25]
```diff
@@ -0,0 +1,181 @@
...
+            cohesity_validate_certs: "{{ var_validate_certs }}"
+            name: CLONE_TASK_NAME
+            view_name: VIEW_NAME
+            environment: VMware
+            vm_names:
+                - VM_Name1
+                - VM_Name2
+            resource_pool: RESOURCE_POOL_NAME
+            state: absent
+```
```
Comment (anvesh-cohesity) [inline docs/tasks/clone_vm.md:154]: verified if this playbook works?

File: docs/tasks/policy.md [hunk 26]
```diff
@@ -0,0 +1,216 @@
...
+            cohesity_policy:
+                name: POLICY_NAME
+                incremental_backup_schedule: ASK_TEAM
+                cluster: cohesity.lab
+                username: admin
+                password: password
+                state: present
+```
+
+### Modify a Protection Policy
```
Comment (anvesh-cohesity) [inline docs/tasks/policy.md]: modifying a policy is not supported

File: docs/tasks/policy.md [hunk 27]
```diff
@@ -0,0 +1,216 @@
...
+    - [Delete a Protection Policy](#Delete-a-Protection-Policy)
+- [How the Task Works](#How-the-Task-works)
+
+## Synopsis
+[top](#Cohesity-policy-management)
+
+Use this task to clone a VM using Test/Dev workflow
+
+#### How It Works
+- The tasks collects all the information (required and optional) and starts a clone task (if *state=present*).
```
Comment (anvesh-cohesity) [inline docs/tasks/policy.md]: this one talks of clone, please change it to policy for this one and the one below

File: docs/tasks/policy.md [hunk 28]
```diff
@@ -0,0 +1,216 @@
...
+- The task can also destory a clone (*state=absent*) that was cloned using Test/Dev workflow.
+
+
+### Requirements
+[top](#Cohesity-policy-management)
+
+* Cohesity DataPlatform running version 6.0 or higher
```
Comment (anvesh-cohesity) [inline docs/tasks/policy.md]: qualified for 6.3 or higher

File: docs/tasks/policy.md [hunk 29]
```diff
@@ -0,0 +1,216 @@
...
+            cohesity_password: "{{ var_cohesity_password }}"
+            cohesity_validate_certs: "{{ var_validate_certs }}"
+            cohesity_policy:
+                name: POLICY_NAME
+                incremental_backup_schedule: ASK_TEAM
+                cluster: cohesity.lab
+                username: admin
+                password: password
+                state: present
+```
```
Comment (anvesh-cohesity) [inline docs/tasks/policy.md]: verified if this playbook works?

File: docs/tasks/policy.md [hunk 30]
```diff
@@ -0,0 +1,216 @@
...
+            cohesity_password: "{{ var_cohesity_password }}"
+            cohesity_validate_certs: "{{ var_validate_certs }}"
+            cohesity_policy:
+                name: POLICY_NAME
+                incremental_backup_schedule: ASK_TEAM
+                cluster: cohesity.lab
+                username: admin
+                password: password
+                state: absent
+```
```
Comment (anvesh-cohesity) [inline docs/tasks/policy.md:156]: verified if this playbook works?

Comment (chandrashekar-cohesity) [reply docs/tasks/policy.md:156]: yes

File: docs/tasks/policy.md (hunk 29 above)
Comment (chandrashekar-cohesity) [reply docs/tasks/policy.md]: udpated and verified yes

File: docs/tasks/clone_vm.md (hunk 25 above)
Comment (chandrashekar-cohesity) [reply docs/tasks/clone_vm.md:154]: yes

File: docs/tasks/clone_vm.md (hunk 24 above)
Comment (chandrashekar-cohesity) [reply docs/tasks/clone_vm.md:114]: Yes

**Comment by chandrashekar-cohesity** [comment]
Will submit a new merge request. Have made changes per comments and updated and verified playbook sample.

**Pull request closed without merging**